sonoshell/README.md

```diff
@@ -0,0 +1,25 @@
+# Sonoshell
+
+This utility provides a mechanism for sonobuoy to run and test a suite of arbitrary commands, reporting the results of each of these back. It is intended to be consumed in a plugin, imported in the Dockerfile as the base image and provided a yaml file defining the suite of commands to run. An example, `test.yaml`, is provided as a guide. By default, sonoshell will run against it, demonstrating the pass and fail conditions, which are based on the exit code of the command, program or script defined for the test. An exit code of 0 indicates that a test has passed, any other exit code is treated as a failure.
+
+Also included is an example plugin definition, `sonoshell.yaml`. This demonstrates a fully functional sonobuoy plugin which reports its results with sonoshell. It is configured to run as a `DaemonSet`, running against and reporting on every node in the cluster, but running it as a `Job` is also possible if the test suite only needs to run on one node.
+
+The example plugin can be tested as follows (assuming running from an environment with a well-configured `sonobuoy` running against an active cluster):
```

Comment:
If you make the requested changes elsewhere you'll want to fixup this explanation since the script can just be included in the yaml. Just edit the script (within the yaml) and run.

Reply:
Updated